patches/monado/0006-distortion-images.patch

```diff
@@ -0,0 +1,28 @@
+diff --git a/src/xrt/compositor/render/render_distortion.c b/src/xrt/compositor/render/render_distortion.c
+index 031f74ad5..1255f2616 100644
+--- a/src/xrt/compositor/render/render_distortion.c
++++ b/src/xrt/compositor/render/render_distortion.c
+@@ -42,7 +42,7 @@ create_distortion_image_and_view(struct vk_bundle *vk,
+ 	    vk,                                                           // vk_bundle
+ 	    extent,                                                       // extent
+ 	    format,                                                       // format
+-	    VK_IMAGE_USAGE_SAMPLED_BIT | VK_IMAGE_USAGE_TRANSFER_DST_BIT, // usage
++	    VK_IMAGE_USAGE_SAMPLED_BIT | VK_IMAGE_USAGE_STORAGE_BIT | VK_IMAGE_USAGE_TRANSFER_DST_BIT, // usage
+ 	    &device_memory,                                               // out_device_memory
+ 	    &image);                                                      // out_image
+ 	VK_CHK_AND_RET(ret, "vk_create_image_simple");
+diff --git a/src/xrt/compositor/shaders/distortion.comp b/src/xrt/compositor/shaders/distortion.comp
+index 08b44d4cf..3e6e468fb 100644
+--- a/src/xrt/compositor/shaders/distortion.comp
++++ b/src/xrt/compositor/shaders/distortion.comp
+@@ -118,8 +118,8 @@ void main()
+ 	vec2 dist_uv = position_to_uv(extent, ix, iy);
+ 
+ 	vec2 r_uv = texture(distortion[iz + 0], dist_uv).xy;
+-	vec2 g_uv = texture(distortion[iz + 2], dist_uv).xy;
+-	vec2 b_uv = texture(distortion[iz + 4], dist_uv).xy;
++	vec2 g_uv = r_uv;
++	vec2 b_uv = r_uv;
```

Comment:
I have doubts the compiler is smart enough to do the `transform_uv` only once as well as `texture(source[iz], r_uv).r,`.

If we only read `r` channel, let's drop the g,b variables from the rest of the shader as well.